backend/controllers/subscriptionController.js

```diff
@@ -0,0 +1,42 @@
+const { UPI_URL } = require("../config/dbConfig");
+const Subscription = require("../model/Subscription");
+
+const postSubscription = async (req, res) => {
+    try {
+      const { userName, plan, startDate, endDate } = req.body;
+      const newSubscription = new Subscription({
+        userName,
+        plan,
+        startDate,
+        endDate
```

Comment:
Shouldn’t we keep a flag to denote if its a charged/free subscription? ‘Coz user will be given free subscription for 1 month and if its a charged subscription we should have payment details like amount, mode, transactionDate etc saved in the db just in case any user raises concern for the same na?

Reply:
@Kajol-Kumari  You're right; including a flag to indicate whether a subscription is free or charged is a good idea. Additionally, storing payment details for charged subscriptions can help resolve any disputes or issues that might arise. I had updated it. can you please check.